### DISCUSSION TITLE
Enforce consistent formatting as part of GitHub Actions workflow

### DISCUSSION THREAD

**Discussion by romanc** [spcl/dace#1804] [Ideas]
Consistent formatting is important to avoid unnecessary conflicts and keep a consistent style in a shared codebase. Contribution guidelines thus kindly ask to run `yapf` before/when submitting a PR. However, consistent formatting isn't enforced in GHA workflows.

Given that two small PRs [^1] [^2] showed formatting changes in code parts that I didn't touch, I would like to start a discussion on automatically enforcing consistent formatting as part of the GitHub Actions workflow.

I am happy to provide a PR for such a workflow early next year assuming we come to an agreement in this discussion.

[^1]: https://github.com/spcl/dace/pull/1731
[^2]: https://github.com/spcl/dace/pull/1803

**Comment by romanc** [comment]
PR to show how this could look like: https://github.com/spcl/dace/pull/1957

**Comment by tbennun** [comment]
Thank you for contributing this work! I think we can close this discussion now as CI now includes this.
